### ISSUE TITLE
C# read-only auto-properties do not warn

### ISSUE THREAD

**Issue by noblethrasher** [dotnet/roslyn#14938]
Version
=====

C# 6.0 with Visual Studio 2015.

Steps to reproduce
============

    sealed class Foo
    {
         public object Bar { get; }

         public Foo()
         {

         }
    }

Expected Behavior
============

Something equivalent to  `warning CS0649: Field (Property) 'Foo.Bar' is never assigned to, and will always have its default value null.`

Actual Behavior
==========

No warning.

**Comment by gafter** [comment]
Since we do not warn on this today, we'd have to enable it under a warning wave.

**Comment by jcouv** [comment]
Depends on https://github.com/dotnet/roslyn/issues/1580 (support for warning waves)

**Comment by CyrusNajmabadi** [comment]
I don't think we can warn on this.  this is a totally normal pattern with things like DI systems.
